src/main/scala/loamstream/googlecloud/CloudSdkDataProcWrapper.scala

```diff
@@ -0,0 +1,92 @@
+package loamstream.googlecloud
+
+import loamstream.util.Loggable
+import loamstream.util.Paths.normalize
+import scala.util.Try
+import scala.util.Success
+import loamstream.util.Tries
+import loamstream.util.ExitCodes
+import scala.concurrent.duration.Duration
+import loamstream.util.BashScript
+import java.nio.file.Paths
+
+/**
+ * @author clint
+ * Nov 28, 2016
+ */
+final class CloudSdkDataProcWrapper private[googlecloud] (
+  config: GoogleCloudConfig,
+  runCommand: Seq[String] => Int = CloudSdkDataProcWrapper.runCommand) extends 
+      DataProcClient.CanStop with DataProcClient.CanTellIfRunning with Loggable {
+
+  import CloudSdkDataProcWrapper.{
+    deleteClusterTokens,
+    isClusterRunningTokens
+  }
+
+  override def stopCluster(): Unit = {
+    debug(s"Deleting cluster '${config.clusterId}'")
+
+    val exitCode = runCommand(deleteClusterTokens(config))
+
+    ExitCodes.throwIfFailure(exitCode)
+  }
+
+  override def isClusterRunning: Boolean = {
+    val exitCode = runCommand(isClusterRunningTokens(config))
+
+    val result = ExitCodes.isSuccess(exitCode)
+
+    debug(s"Cluster '${config.clusterId}' running? $result")
+
+    result
+  }
+}
+
+object CloudSdkDataProcWrapper extends Loggable {
+  def fromConfig(config: GoogleCloudConfig): Try[CloudSdkDataProcWrapper] = {
+    val gcloudBinary = config.gcloudBinary.toFile
+
+    if (gcloudBinary.exists && gcloudBinary.canExecute) {
+      Success(new CloudSdkDataProcWrapper(config))
+    } else {
+      Tries.failure(s"gcloud executable not found at ${config.gcloudBinary} or not executable")
+    }
+  }
+
+  private[googlecloud] def deleteClusterTokens(config: GoogleCloudConfig): Seq[String] = {
+    gcloudTokens(config)("delete")(config.clusterId)
+  }
+
+  private[googlecloud] def isClusterRunningTokens(config: GoogleCloudConfig): Seq[String] = {
+    gcloudTokens(config)("describe")(config.clusterId)
+  }
+  
+  private[googlecloud] def gcloudTokens(config: GoogleCloudConfig)(verb: String)(args: String*): Seq[String] = {
+    val gcloud = normalize(config.gcloudBinary)
+
+    gcloud +: "beta" +: "dataproc" +: "clusters" +: verb +: "--project" +: config.projectId +: args
+  }
+
+  private def runCommand(tokens: Seq[String]): Int = runProcess(tokens.mkString(" "), "Google Cloud SDK")
```

Comment:
Just noting from previous comment that now that I see tokens is `Seq[String]` and so something like `tokens.runCommand()` wouldn't be possible unless it was its own type... which might not be the worst idea for type-safety?

```scala
final case class GCloudTokens(tokens: Seq[String])
```

Reply:
I see what you mean, and I'll think it over.  My gut feeling is that the benefit isn't very big here.  This would mean (maybe) one more way of running commands.  There are already 2 or 3 depending on how you count, and I'm already trying to consolidate _those_ into one way.

Reply:
I liked this idea, but it didn't provide a lot of bang for the buck when I tried it.  :\  I'll revisit this the next time I work on this code.